### PR TITLE
Support more monit config options

### PR DESCRIPTION
**Pull request by cwalsh**
* Supports [multiple groups](https://mmonit.com/monit/documentation/monit.html#SERVICE-GROUPS) in the monit template
* Fixes tests
* Tests now also support Ruby 2.2
* Supports [Service Dependencies](https://mmonit.com/monit/documentation/monit.html#SERVICE-DEPENDENCIES)

### CODE REVIEW

File: spec/defines/monit_monitor_spec.rb [hunk 1]
```diff
@@ -75,13 +75,34 @@
       it 'should compile' do
         should contain_file('/etc/monit.d/monit-monitor-group.conf').with_content(
           "check process monit-monitor-group with pidfile /var/run/monit.pid\n" +
-          "  start program = \"/etc/init.d/monit-monitor-group start\"\n" +
-          "  stop program  = \"/etc/init.d/monit-monitor-group stop\"\n" +
+          "  start program = \"/usr/sbin/invoke-rc.d monit-monitor-group start\"\n" +
+          "  stop program  = \"/usr/sbin/invoke-rc.d monit-monitor-group stop\"\n" +
```
Comment (cwalsh) [inline spec/defines/monit_monitor_spec.rb]: These specs should have been changed when the template was changed. Oh well, fixed now.

**Comment by bc-vincent-zhao** [comment]
I'm not ruby experts, but the changes in this PR seems straight forward enough. 👍

**Comment by cwalsh** [comment]
Thanks Vincent. @dsolsona or @chrisboulton are you able to take a look at this, too, please?

**Comment by dsolsona** [comment]
lgtm 👍 

And it even has tests ;) so you get another 👍

**Pull request merged**